### ISSUE TITLE
Help Installing on MacOS for newbies

### ISSUE THREAD

**Issue by ZheShiSheiAh** [Noah-C-S/SlippiMusic#6]
Trying to install on MacOS, following the instructions installing through the source code - all was going well, until I go to run [python3 slippiMusic.py]. I get met with [ModuleNotFoundError: No module named 'enet']. Now I know literally nothing about coding, but I've got no idea what this is, how to fix it, and google isn't helping me much either:( 

Sorry if this is the wrong place to be posting this, if there's a more suitable way to discuss or get in touch, please let me know.

**Comment by Noah-C-S** [comment]
enet is a library that I use to communicate with Slippi while it runs. Hopefully, this problem can be solved just by opening up a terminal and executing `pip3 install pyenet`, which will install the required module. You'll probably need to execute `pip3 install pygame` as well. Let me know if that fixes it.

**Comment by ZheShiSheiAh** [comment]
You're gonna have to be patient with me, but I tried running `pip3 install pyenet` again, and I just got a huge wall of red text :/

This could be an issue with my Python, as I'm not actually sure if I installed it properly!

Here's what it says if its any use: 
Running setup.py install for pyenet ... error
    ERROR: Command errored out with exit status 1:
     command: /Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/bb/nj505z216nnfsdmth73llhnw0000gn/T/pip-install-8q5w3pkd/pyenet/setup.py'"'"'; __file__='"'"'/private/var/folders/bb/nj505z216nnfsdmth73llhnw0000gn/T/pip-install-8q5w3pkd/pyenet/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/bb/nj505z216nnfsdmth73llhnw0000gn/T/pip-record-i2r280sw/install-record.txt --single-version-externally-managed --compile --install-headers /Library/Frameworks/Python.framework/Versions/3.9/include/python3.9/pyenet
         cwd: /private/var/folders/bb/nj505z216nnfsdmth73llhnw0000gn/T/pip-install-8q5w3pkd/pyenet/
    Complete output (12 lines):
    running install
    running build
    running build_ext
    Compiling enet.pyx because it depends on /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/Cython/Includes/libc/string.pxd.
    [1/1] Cythonizing enet.pyx
    building 'enet' extension
    creating build
    creating build/temp.macosx-10.9-x86_64-3.9
    creating build/temp.macosx-10.9-x86_64-3.9/enet
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch x86_64 -g -DHAS_POLL -DHAS_FCNTL -DHAS_MSGHDR_FLAGS -DHAS_SOCKLEN_T -Ienet/include/ -I/Library/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c enet.c -o build/temp.macosx-10.9-x86_64-3.9/enet.o -O3
    xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
    error: command '/usr/bin/gcc' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/bb/nj505z216nnfsdmth73llhnw0000gn/T/pip-install-8q5w3pkd/pyenet/setup.py'"'"'; __file__='"'"'/private/var/folders/bb/nj505z216nnfsdmth73llhnw0000gn/T/pip-install-8q5w3pkd/pyenet/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/bb/nj505z216nnfsdmth73llhnw0000gn/T/pip-record-i2r280sw/install-record.txt --single-version-externally-managed --compile --install-headers /Library/Frameworks/Python.framework/Versions/3.9/include/python3.9/pyenet Check the logs for full command output.

**Comment by Noah-C-S** [comment]
All right, I'm not sure what most of that means, but enet depends on some C code and uses gcc to compile it. It says that it's `missing xcrun at /Library/Developer/CommandLineTools/usr/bin/xcrun`, and a quick Google search indicates that this can be fixed by installing the xcode toolkit, which seems to be a terminal command. Hopefully, this one will work: [Here's what I found on installing xcode toolkit](https://ma.ttias.be/mac-os-xcrun-error-invalid-active-developer-path-missing-xcrun/)

**Comment by ZheShiSheiAh** [comment]
Ok those commands didn't work either, but I'm trying to install manually from the Apple Developer downloads so I'll get back to you when I've managed that. I really appreciate the help, I understand this is probably like pulling teeth!

**Comment by ZheShiSheiAh** [comment]
So as far as I know, I've installed the Xcode Toolkit, but trying `pip3 install pyenet` gives back the same red wall, with the same `missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun` part.

**Comment by Noah-C-S** [comment]
Hmmm, try going to the folder specified by that error message, if there's nothing there, then I would say that Xcode Toolkit isn't installed. 

Also, what error message does `xcode-select --install` produce? Does it just say that xcode-select doesn't exist? I'm not very familiar with MacOS so if you can't get pip to work, then there isn't that much I can do. I think Xcode (which is an IDE) is also on the App store; though it's a larger install, I think it will come with the toolkit so you can try installing that. 

You could try generic things like restarting your computer, updating your version of MacOS (I only know this to be working on the newest version when I tested it, which was Big Sur) or you could contact Apple support and tell them that you need to install Xcode toolkit for pip and it's not working.  Other than that, I'm out of ideas, unless you have more information.

**Comment by ZheShiSheiAh** [comment]
Okay well you were right in that when did "Go: `/Library/Developer/CommandLineTools/usr/bin/xcrun`", there was no `bin` in `usr`. But uh, `xcode-select --install` didn't give any error message, it downloaded some software! I've now got a `bin`, but no `xcrun` in it.

You reckon I should just try `pip3 install pyenet` again?

**Comment by Noah-C-S** [comment]
Sure. If that still doesn't work, you could try `sudo xcode-select --reset` and then try installing it again.

**Comment by ZheShiSheiAh** [comment]
Sick! After picking up with the instructions I could follow it through to the end this time! Slippi is running with music as I type :D 

A tip for any MacOS users that might be reading this, if `cd absolute/path/to/folder/slippiMusic.py/is/located` is coming back with "No such file or directory", then just try this. Type `cd` in the terminal, then open a finder window, then go wherever you've got the SlippiMusic-1.2.1 folder sitting (for me I have a **folder** in my **applications folder** named _Melee_, with Slippi Dolphin, my User.json and Melee ISO in), and then just drag and drop the SlippiMusic-1.2.1 folder into the terminal window! Seems pretty self explanatory but I had no idea until I found it after like 2 hours of googling lol.

After that just follow the rest of the instructions and hopefully it will work, or at least that's what I managed to do.

Thanks so much for your help Noah!
